### ISSUE TITLE
Debugger's tokenPos mapping doesn't seem to match up

### ISSUE THREAD

**Issue by DanTup** [dart-lang/sdk#32734]
Given the following source code:

```dart
main() async {
  var a = [1, 2, 3, 4, 5];
  var b = a.where((i) => i % 2 == 0);
  print(b);
}
```

I sent a breakpoint for the expression `i % 2` (line 3, column 28):

```js
{
	"id": "20",
	"method": "addBreakpointWithScriptUri",
	"params": {
		"isolateId": "isolates/403546744",
		"scriptUri": "/Users/dantup/Desktop/Dart%20Sample/bin/main.dart",
		"line": 3,
		"column": 28
	}
}
```

I got back a response that also shows `3:28`:

```js
{
	"jsonrpc": "2.0",
	"method": "streamNotify",
	"params": {
		"streamId": "Debug",
		"event": {
			"type": "Event",
			"kind": "BreakpointAdded",
			"isolate": {
				"type": "@Isolate",
				"fixedId": true,
				"id": "isolates\/403546744",
				"name": "main.dart:main()",
				"number": "403546744"
			},
			"timestamp": 1522672467360,
			"breakpoint": {
				"type": "Breakpoint",
				"fixedId": true,
				"id": "breakpoints\/1",
				"breakpointNumber": 1,
				"resolved": false,
				"location": {
					"type": "UnresolvedSourceLocation",
					"script": {
						"type": "@Script",
						"fixedId": true,
						"id": "libraries\/@17142085\/scripts\/file%3A%2F%2F%2FUsers%2Fdantup%2FDesktop%2FDart%2520Sample%2Fbin%2Fmain.dart\/1628659fd71",
						"uri": "file:\/\/\/Users\/dantup\/Desktop\/Dart%20Sample\/bin\/main.dart",
						"_kind": "script"
					},
					"line": 3,
					"column": 28
				}
			}
		}
	}
}
```

Later, I got a resolved breakpoint with a `tokenPos` of `48`:

```js
{
	"jsonrpc": "2.0",
	"method": "streamNotify",
	"params": {
		"streamId": "Debug",
		"event": {
			"type": "Event",
			"kind": "BreakpointResolved",
			"isolate": {
				"type": "@Isolate",
				"fixedId": true,
				"id": "isolates\/403546744",
				"name": "main.dart:main()",
				"number": "403546744"
			},
			"timestamp": 1522672467374,
			"breakpoint": {
				"type": "Breakpoint",
				"fixedId": true,
				"id": "breakpoints\/1",
				"breakpointNumber": 1,
				"resolved": true,
				"location": {
					"type": "SourceLocation",
					"script": {
						"type": "@Script",
						"fixedId": true,
						"id": "libraries\/@17142085\/scripts\/file%3A%2F%2F%2FUsers%2Fdantup%2FDesktop%2FDart%2520Sample%2Fbin%2Fmain.dart\/1628659fd71",
						"uri": "file:\/\/\/Users\/dantup\/Desktop\/Dart%20Sample\/bin\/main.dart",
						"_kind": "script"
					},
					"tokenPos": 48
				}
			}
		}
	}
}
```

However, if I'm reading the token table correctly, it says that `tokenPos: 48` is at `4:3`:

```js
{
	"jsonrpc": "2.0",
	"result": {
		"type": "Script",
		"class": {
			"type": "@Class",
			"fixedId": true,
			"id": "classes\/15",
			"name": "Script"
		},
		"size": 112,
		"fixedId": true,
		"id": "libraries\/@17142085\/scripts\/file%3A%2F%2F%2FUsers%2Fdantup%2FDesktop%2FDart%2520Sample%2Fbin%2Fmain.dart\/1628659fd71",
		"uri": "file:\/\/\/Users\/dantup\/Desktop\/Dart%20Sample\/bin\/main.dart",
		"_kind": "script",
		"_loadTime": 1522672467313,
		"library": {
			"type": "@Library",
			"fixedId": true,
			"id": "libraries\/@17142085",
			"name": "",
			"uri": "file:\/\/\/Users\/dantup\/Desktop\/Dart%20Sample\/bin\/main.dart"
		},
		"lineOffset": 0,
		"columnOffset": 0,
		"source": "main() async {\n  var a = [1, 2, 3, 4, 5];\n  var b = a.where((i) => i % 2 == 0);\n  print(b);\n}\n",
		"tokenPosTable": [
			[
				[1,0,1,1,5,2,6,3,8,4,14],
				[2,6,3,7,7,8,9,9,11,10,12,11,13,12,15,13,16,14,18,15,19,16,21,18,22,19,24,21,25,22,26],
				[3,24,3,25,7,27,9,28,11,29,12,30,13,32,18,33,19,34,20,36,21,37,23,38,26,40,28,41,30,42,32,43,35,45,36,46,37],
				[4,48,3,50,8,51,9,53,10,54,11],
				[5,56,1]
			]
		]
	},
	"id": "24"
}
```

**Comment by DanTup** [comment]
@kevmoo This is the debugger/VM/observatory, not analyzer. This is when I'm adding breakpoints and stepping through code, the debugger seems to return bogus info for the positions in the source code.

**Comment by kevmoo** [comment]
@DanTup – ack

**Comment by ZichangG** [comment]
@DanTup I can't reproduce the problem with VScode.
vm: `Dart VM version: 2.2.1-edge.fd48001ace793b4abeb1c9cc3803da25c73c52b4 (Mon Apr 22 15:31:09 2019 +0000) on "linux_x64"`
Here are some printings I put inside debugger running the script you provided:
`breakpoint location: (line: 3 , col: 28), token range: (42 to 79)`
`Resolve break point: token position: 69`

Call stack when hitting the breakpoint:
main.<anonymous closure> (/usr/local/google/home/zichangguo/dart-sdk/sdk/Junk/main.dart:3)
WhereIterator.moveNext (dart:_internal/iterable.dart:439)
_iterablePartsToStrings (dart:collection/iterable.dart:318)
IterableBase.iterableToShortString (dart:collection/iterable.dart:238)
Iterable.toString (dart:core/iterable.dart:667)
_StringBase._interpolateSingle (dart:core-patch/string_patch.dart:821)
print (dart:core/print.dart:9)
main (/usr/local/google/home/zichangguo/dart-sdk/sdk/Junk/main.dart:4)
<asynchronous gap> (Unknown Source:0)
main (/usr/local/google/home/zichangguo/dart-sdk/sdk/Junk/main.dart:0)
_Closure.call (dart:core-patch/function.dart:0)
_startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:300)
_Closure.call (dart:core-patch/function.dart:0)
_RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:171)

**Comment by DanTup** [comment]
@ZichangG looks like this may be recently fixed? I can repro with 2.2.1-dev.3 (I get a token pos that still points at the start of line 4), however in the most recent nightly (`2.2.1-edge.a58aa6f705d6467fd1574bee33b11d6d815e12dd (Tue Apr 23 08:58:58 2019 +0000)`) I get location on line 3 where I expect (and the debugger now visibly breaks on line 3 instead of line 4).
